### DISCUSSION TITLE
Old values vs. new values onchange

### DISCUSSION THREAD

**Discussion by moussa1** [sa-si-dev/virtual-select#40] [Q&A] [answered]
Is there any way onchange to know what value options have been deselected/selected?

**Comment by sa-si-dev** [comment, marked as answer]
@moussa1 Virtual select dosen't have any feature like this, but you could create your own callback function for onChange event to find differents between old and new values.

**Reply by bes-internal** [reply]
and use [getSelectedOptions()](https://sa-si-dev.github.io/virtual-select/#:~:text=Copied-,getSelectedOptions(),-Get%20selected%20option%27s)

**Comment by kjmitch** [comment]
Could I have some clarification, @sa-si-dev?

Your answer to this question was that there was no programmatic way to query a VirtualSelect object and get the currently-selected item's value (or multiple items' values in an array). Is that still the case? @bes-internal's reply  seems to indicate either that the `getSelectedOptions()` method was added to implement this functionality, or that this method would be needed when implementing my own `onChange` callback as you suggested. Is my understanding correct?

**Reply by sa-si-dev** [reply]
The `getSelectedOptions()` would return array of object. I don't think that has anything to do with your case. You could use element value itself for your implementation.

**Reply by kjmitch** [reply]
Apologies, I forgot to follow up on this. I did see that using the `.value` property for a menu element returns the set of values that are currently selected in the menu. Thanks so much for your help!
